src/UI/Buyer/src/app/services/auth/auth.service.ts

```diff
@@ -134,6 +140,16 @@ export class AuthService {
       false,
       rememberMe
     )
+
+    let user = await Me.Get({ accessToken: creds.access_token })
+    if (user.xp?.Language) {
+      this.cookieService.putObject(this.languageCookieName, user.xp?.Language)
+      this.translate.use(user.xp?.Language);
+    } else {
+      this.cookieService.remove(this.languageCookieName)
+      this.translate.use(this.translate.defaultLang);
```

Comment:
Shouldn't the logic here be similar to where we try to use the user's location to retrieve the most relevant language?

Reply:
Agreed, I have been working creating a shared service specifically to manage getting and setting the language to adhere to the single responsibility principle.

Reply:
I have moved this implementation into language-selector.service.ts. One for the buyer app and one for the seller app as each app has their own specific services, helpers, etc. that are not reusable.